### PR TITLE
Use revision tokens in generate service

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-884

This involved a bit of a refactor to support testability.

**Release Note**

```release-note
Support revision tokens in generate service
```

/assign @mikehelmick

### CODE REVIEW

File: internal/generate/handler.go [hunk 1]
```diff
@@ -34,166 +36,160 @@ import (
...
+	}
+
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprint(w, "successfully generated exposure keys")
+}
+
+func (h *generateHandler) generate(ctx context.Context, regions []string) error {
+	logger := h.logger.Named("generate")
+
+	if h.config.KeysPerExposure < 2 {
```
Comment (mikehelmick) [inline internal/generate/handler.go:99]: why 2?

Comment (sethvargo) [reply internal/generate/handler.go:99]: Line 141:

```go
intervalIdx, err := util.RandomInt(len(publish.Keys) - 1)
```

That gets a random index in the list, but panics if the number of keys is 1 (and therefore tries to generate a random int between 0 and 0).

Comment (mikehelmick) [reply internal/generate/handler.go:99]: worth a comment in the code then

**Pull request merged**